src/utils/objectUtils.ts

```diff
@@ -123,3 +123,25 @@ export function groupByUnique<
     {} as Record<RecordKeyType, T>,
   )
 }
+
+type DatesAsString<T> = T extends Date ? string : ExactlyLikeWithDateAsString<T>
+
+type ExactlyLikeWithDateAsString<T> = T extends object ? { [K in keyof T]: DatesAsString<T[K]> } : T
+
+export function convertDatesToIsoString<Input extends object>(
```

Comment:
naming proposal: convertDateFieldsToIsoString

Reply:
Yeah! Your idea is way more clear! I will modify it, thanks Igor 🙇